inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -1618,12 +1789,159 @@ void MKLDNNReduceNode::reduce_type(const uint8_t *in_ptr, uint8_t *out_ptr, size
         reduce_PLN(in_ptr, out_ptr);
     } else {
         if ((algorithm == ReduceAnd || algorithm == ReduceLogSumExp || algorithm == ReduceMax ||
-             algorithm == ReduceMin || algorithm == ReduceProd) && ReduceC) {
+             algorithm == ReduceMin || algorithm == ReduceProd) && ReduceC && (IC % blk_size)) {
             reduce_BLK_concern_padding(in_ptr, out_ptr);
         } else {
             reduce_BLK(in_ptr, out_ptr);
         }
     }
+
+    if (is_hybrid_layout) {
+        uint8_t *proc_ptr = out_ptr;
+        auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+        out_ptr = reinterpret_cast<uint8_t *>(dstMemPtr->GetPtr());
+        if (is_nspc) {
+            nspc2ncsp(proc_ptr, out_ptr);
+        } else {
+            blocked2ncsp(proc_ptr, out_ptr);
```

Comment:
From the duplication perspective, I totally agree. Yet the main concern is that for most cases reorder layer calls the oneDNN implementation, which as I understand is a tensor to tensor routines. But we need to reuse buffer to buffer routines, about which only a uint8 optimizedNcsp2Nspc routine exists in reorder layer. Further discussions are always welcomed. Thanks Maksim!

Reply:
Technically it is possible to construct auxiliary oneDNN reorder primitives if necessary on the `createPrimitive()` stage, since on that stage all the necessary memory addresses and tensor descriptors are known, and use them here. Because oneDNN library can provide good jitted implementations for some combinations. But it is also known that there are cases where such a simple c++ implementation can be good enough or even better. Have you estimated which is better to use here oneDNN reorder or this  c++ implementation. I mean is it worse the effort to reuse oneDNN reorder primitives here?